app/controllers/agent/ManageAgentController.scala

```diff
@@ -142,59 +159,128 @@ class ManageAgentController @Inject()(
                                        )
       ipPropertyLinksCount = propertyLinks.total
     } yield {
-      (ipPropertyLinksCount, agentToBeManagedOpt) match {
-        case (0, Some(agent)) if agent.propertyCount == 0 =>
-          //IP has no property links but still has an agent
-          Some(
-            removeAgentFromOrganisationView(
-              submitAgentAppointmentRequest,
-              agent.representativeCode,
-              agent.name,
-              calculateBackLink(organisationsAgents, agent.representativeCode)))
-        case (1, Some(agent)) if agent.propertyCount == 0 =>
-          //IP has one property link but agent is not assigned
-          Some(
-            manageAgentView(
-              submitManageAgentForm,
-              ManageAgentOptions.onePropertyLinkNoAssignedAgentsOptions,
-              agent,
-              calculateBackLink(organisationsAgents, agent.representativeCode)))
-        case (1, Some(agent)) if agent.propertyCount == 1 =>
-          //IP has one property link and agent is assigned to that property
-          Some(
-            unassignAgentFromPropertyView(
-              submitAgentAppointmentRequest,
-              agent,
-              calculateBackLink(organisationsAgents, agent.representativeCode)))
-
-        case (numberOfPropertyLinks, Some(agent)) if numberOfPropertyLinks > 1 && agent.propertyCount == 0 =>
-          //IP has more than one property links but agent is not assigned to any
-          Some(
-            manageAgentView(
-              submitManageAgentForm,
-              ManageAgentOptions.multiplePropertyLinksNoAssignedAgentsOptions,
-              agent,
-              calculateBackLink(organisationsAgents, agent.representativeCode)
-            ))
+      if (featureSwitch.isAgentListYearsEnabled) {
+        val ManageAgentOptionItemList =
+          (ipPropertyLinksCount, agentToBeManagedOpt) match {
```

Comment:
There's not a case here for when agentToBeManagedOpt is None. It is probably better to do that check above before the feature switch check, as if the user is on this page without an agent then nothing should work, regardless of feature switch or how many properties they have.

Reply:
Code has been updated, branch rebased.

Reply:
Should I do `case None => None` or case _ => None ??